src/frontend/src/pages/AdminToolsPage/AdminToolsUserManagement.tsx

```diff
@@ -0,0 +1,142 @@
+import LoadingIndicator from '../../components/LoadingIndicator';
+import ErrorPage from '../ErrorPage';
+import { fullNamePipe } from '../../utils/pipes';
+
+interface UserData {
+  userId: number;
+  role: string;
+}
+
+const AdminToolsUserMangaement: React.FC = () => {
+  const [role, setRole] = useState('');
+  const [user, setUser] = useState<UserData | null>(null);
+  const [isDisabled, setIsDisabled] = useState(true);
+  const [hideSuccessLabel, setHideSuccessLabel] = useState(true);
+  const { isLoading, isError, error, data } = useAllUsers();
+  const update = useUpdateUserRole();
+
+  if (isLoading || !data) return <LoadingIndicator />;
+
+  if (isError) return <ErrorPage message={error?.message} />;
+
+  const handleSearchChange = (event: React.SyntheticEvent<Element, Event>, value: string | null) => {
+    if (value) {
+      const user = data.find((user) => fullNamePipe(user) === value.split(' -')[0]);
+      if (user) setUser(user);
+    } else {
+      setUser(null);
+    }
+  };
+
+  const handleRoleChange = (event: SelectChangeEvent) => {
+    setRole(event.target.value);
+    if (user && event.target.value === user.role) {
+      setIsDisabled(true);
+    } else {
+      setIsDisabled(false);
+    }
+  };
+
+  const handleClick = async () => {
+    setHideSuccessLabel(true);
+    await update.mutateAsync({ userId: user?.userId, role }).catch((error) => {
+      alert(error);
+      throw new Error(error);
+    });
+    setHideSuccessLabel(false);
+  };
+
+  return (
+    <PageBlock title={'Role Management'}>
+      <Grid container spacing={2}>
+        <Grid item xs={12} md={8}>
+          <Autocomplete
+            disablePortal
+            id="autocomplete"
+            onChange={handleSearchChange}
+            options={data.map((user) => `${fullNamePipe(user)} - ${user.email}`)}
+            sx={{
+              height: '40px',
+              backgroundColor: 'ActiveBorder',
+              width: '100%',
+              borderRadius: '25px',
+              borderColor: 'black',
+              '.MuiOutlinedInput-notchedOutline': {
+                border: 0,
+                borderRadius: '25px'
+              }
+            }}
+            size="small"
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                InputProps={{
+                  ...params.InputProps,
+                  startAdornment: (
+                    <InputAdornment position="start">
+                      <SearchIcon />
+                    </InputAdornment>
+                  )
+                }}
+                placeholder="Select a User"
+              />
+            )}
+          />
+        </Grid>
+        <Grid item xs={12} md={4}>
+          <FormControl
+            size="small"
+            sx={{
+              width: '100%',
+              backgroundColor: '#ff0000',
+              borderColor: '#0062cc',
+              borderRadius: '25px',
+              height: '40px'
```

Comment:
make a style object for this one two (in general if the css is getting into multiple lines, makes a style object)